src/riscv_cpu.cpp

```diff
@@ -1729,6 +1737,12 @@ static void handle_mret(RISCVCPUState *s)
     int mpp = (s->mstatus & MSTATUS_MPP) >> MSTATUS_MPP_SHIFT;
     s->mstatus &= ~MSTATUS_MPP;
 
+    // Clear MPRV on non m-mode mret
```

Comment:
I can't find this in the spec. Can you point me to the section that explains it?

Reply:
It's in the 1.12 spec, page 21

![image](https://user-images.githubusercontent.com/2322266/83467447-e8960300-a447-11ea-985e-1fea848995de.png)

Reply:
I see. My older copy doesn't have the last part.
I'm curious what piece of code triggered that mismatch because we clear MPRV in the software trap handler after emulating the instruction.

Reply:
riscv-dv